frontend/src/metabase/components/ListSearchField/ListSearchField.tsx

```diff
@@ -0,0 +1,39 @@
+import { useEffect, useRef } from "react";
+
+import { ActionIcon, Icon, TextInput, type TextInputProps } from "metabase/ui";
+
+// eslint-disable-next-line import/no-default-export -- deprecated usage
+export default function ListSearchField({
+  autoFocus,
+  ...props
+}: TextInputProps & { onResetClick?: () => void }) {
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
```

Comment:
I'm not touching this part of the code for now - since `TextInput` is fundamentally a `input` this still works for now